modules/grandpa/src/lib.rs

```diff
@@ -130,7 +130,7 @@ pub mod pallet {
 		))]
 		pub fn submit_finality_proof(
 			origin: OriginFor<T>,
-			finality_target: BridgedHeader<T, I>,
+			finality_target: Box<BridgedHeader<T, I>>,
 			justification: GrandpaJustification<BridgedHeader<T, I>>,
```

Comment:
Isn't `Justification` bigger than the header?

Reply:
No, it keeps most of stuff on heap, so only `sizeof(Vec<>)` is accounted ~64bits.